sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/persons/PersonDirectorySteps.java

```diff
@@ -64,38 +71,136 @@ public PersonDirectorySteps(
           webDriverHelpers.isElementVisibleWithTimeout(UUID_INPUT, 20);
         });
 
-    When(
-        "I navigate to the last created Person page via URL",
+    Then(
+        "I choose random value for Year of birth filter",
         () -> {
-          String createdPersonUUID = EditContactPersonSteps.fullyDetailedPerson.getUuid();
-          String LAST_CREATED_PERSON_PAGE_URL =
-              environmentUrl + "/sormas-webdriver/#!persons/data/" + createdPersonUUID;
-          webDriverHelpers.accessWebSite(LAST_CREATED_PERSON_PAGE_URL);
+          String yearOfBirth = apiState.getLastCreatedPerson().getBirthdateYYYY().toString();
           webDriverHelpers.waitForPageLoaded();
-          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(UUID_INPUT, 50);
+          webDriverHelpers.selectFromCombobox(BIRTH_YEAR_COMBOBOX, yearOfBirth);
         });
 
-    When(
-        "I search for specific person in person directory",
+    Then(
+        "I choose random value for Month of birth filter",
         () -> {
-          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(
-              SEARCH_PERSON_BY_FREE_TEXT, 30);
-          final String personUuid = EditEventSteps.person.getUuid();
-          webDriverHelpers.fillAndSubmitInWebElement(SEARCH_PERSON_BY_FREE_TEXT, personUuid);
+          String monthOfBirth = apiState.getLastCreatedPerson().getBirthdateMM().toString();
+          webDriverHelpers.selectFromCombobox(BIRTH_MONTH_COMBOBOX, monthOfBirth);
         });
 
-    When(
-        "I click on specific person in person directory",
+    Then(
+        "I choose random value for Day of birth filter",
```

Comment:
add also the location of the action since this filter is available in multiple places in the application